src/Core/PluginCollection.php

```diff
@@ -385,4 +396,20 @@ public function with(string $hook): Generator
             }
         }
     }
+
+    /**
+     * @return \Cake\Core\ContainerInterface
+     */
+    public function getContainer(): ContainerInterface
```

Comment:
Do we need this? If we don't need to externally fetch the container lets not expand our public API too far. If there is a need for it we can discuss the use case when it comes up.